aws-devicefarm-gradle-plugin/src/main/groovy/com.amazonaws.devicefarm/DeviceFarmServer.java

```diff
@@ -108,6 +110,19 @@ public void uploadApks(final String variantName, final File testPackage, final F
                 .withFilter(extension.getTest().getFilter())
                 .withTestPackageArn(uploadTestPackageIfNeeded(project, testPackage));
 
+        if (!extension.getExecutionConfiguration().getVideoRecording()) {
+            runTest.addParametersEntry("video_recording", "false");
+        }
+
+        if (!extension.getExecutionConfiguration().getPerformanceMonitoring()) {
+            runTest.addParametersEntry("app_performance_monitoring", "false");
+        }
+
+        if (extension.getTest().getTestType().equals(TestType.APPIUM_JAVA_JUNIT) || extension.getTest().getTestType().equals(TestType.APPIUM_JAVA_TESTNG) || extension.getTest().getTestType().equals(TestType.APPIUM_PYTHON)) {
```

Comment:
Would break this into a separate private static method to determine if its an Appium test type or not.